src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -168,6 +171,27 @@ void AssertAndPersistAnalyzer(Tree configurationTree, string configurationKey, s
 
                     configurationTree.Add(configurationKey, expectedAnalyzer);
                 }
+
+                void PersistSearchEngine(Tree configurationTree,  string configurationKey, SearchEngineType defaultEngineType)
```

Comment:
pass Index here and calculate everything inside the method. The `configurationKey` is incorrect for exception message. Should use RavenConfiguration.GetKey there.

Reply:
done